### ISSUE TITLE
Use of Mailbox and string is inconsistent and not thought-through

### ISSUE THREAD

**Issue by Laeeth** [symmetryinvestments/imap-d#12]

**Comment by otrho** [comment]
I think this is at least partly referring to the mailbox hierarchical separator which we don't seem to support properly.

The character used **is defined by the server instance**, and is usually `.` or `/` but needs to be used properly.  Our mailbox handling and specification needs to support this more generally.

E.g., Right now we can perform mailbox operations using `Mailbox` objects which may have names containing `.` or `/` and they'll be passed through to the server as is and probably not work as expected.

**Comment by Laeeth** [comment]
Yes. I just ran out of time before implementing those and utf7 mailbox names properly

**Comment by otrho** [comment]
OK, also sending strings should be abstracted as there are quoted "strings" and there are string literals:
```
{33}
potentially
multiline
strings
```
We should have a sendString method which could like a slice of strings and send them as a string literal.

**Comment by otrho** [comment]
Also -- not necessarily related to this issue but is another basic feature we should support -- IDs can be UIDs, sequence IDs and also 'sequence sets', e.g., `2:4` meaning sequence IDs 2, 3 and 4.
